### PR TITLE
95 cli step 5 b04 anglepy

### PR DESCRIPTION
**Pull request by cpaniaguam**
TODO: choose better name? @mochell 
#95

### CODE REVIEW

**Review by hollandjg — CHANGES_REQUESTED** [review]
Thanks for this! I've got a few more comments here.

File: src/icesat2_tracks/analysis_db/B04_angle.py [hunk 1]
```diff
@@ -33,872 +32,905 @@
...
+            return mask, k, dd_rm, np.arange(0, mask.size)[mask]
 
-    print("k_upper_lim ", k_upper_lim)
 
-    # variance method
-    amp_data = np.sqrt(GGk.gFT_cos_coeff**2 + GGk.gFT_sin_coeff**2)
-    mask, k, weights, positions = define_wavenumber_weights_tot_var(
-        amp_data, m=1, k_upper_lim=k_upper_lim, variance_frac=0.20, verbose=False
-    )
+        def plot_instance(
```
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B04_angle.py:407]: Indeed, that function, and a number of others in this file, do use variables defined in the scope.  Fixing this entails significant effort. It's almost a `wont-fix` for me.

File: src/icesat2_tracks/analysis_db/B04_angle.py [hunk 2]
```diff
@@ -33,872 +32,905 @@
...
-        markersize=2,
-        linewidth=0.8,
-    )
+            plt.xlim(min([-90, np.nanmin(dir_best)]), max([np.nanmax(dir_best), 90]))
 
-plt.ylabel("Density")
-plt.title("weight margins", loc="left")
 
-plt.xlim(-90, 90)
+            ax3 = F.fig.add_subplot(gs[2, 0:-1]) # can the assignment be removed? CP
```
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B04_angle.py]: It's hard to say if it'll work the same way without granular tests. I think `add_subplot` creates a side effect and returns `None` but I'm speculating.

File: src/icesat2_tracks/analysis_db/B04_angle.py [hunk 3]
```diff
@@ -33,872 +32,905 @@
...
+                "alpha": (
+                    Prior_smth.sel(k=k_prime_max, method="nearest").Prior_direction.data,
+                    Prior_smth.sel(k=k_prime_max, method="nearest").Prior_spread.data,
                 )
-                F.ax3.axhline(
-                    prior_sel["alpha"][0] - prior_sel["alpha"][1],
-                    color="green",
-                    linewidth=0.7,
+            }
+            SM.fitting_kargs = {"prior": prior_sel, "prior_weight": 3} # not sure this might be used somewhere. CP
```
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B04_angle.py]: Not easy to say. I'll just leave there for now.

File: src/icesat2_tracks/analysis_db/B04_angle.py [hunk 4]
```diff
@@ -20,8 +19,8 @@
 
 
 from matplotlib.gridspec import GridSpec
-
-from numba import jit
+import matplotlib.pyplot as plt
+from numba import jit # maybe for later optimizations?  # noqa: F401
```
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B04_angle.py]: I'll comment the line to keep it as a reminder and avoid loading the unused module.

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py [hunk 5]
```diff
@@ -41,11 +41,12 @@ def run_A02c_IOWAGA_thredds_prior(
     batch_key: str = typer.Option(..., callback=validate_batch_key),
     ID_flag: bool = True,
     output_dir: str = typer.Option(None, callback=validate_output_dir),
+    verbose: bool = False
 ):
     """
     TODO: add docstring
     """
-    with suppress_stdout():
+    with suppress_stdout(verbose):
```
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:49]: I was intending to suppress all the output! Note that the rest of the code is within the scope of the context manager.

**Review by kmilo9999 — COMMENTED** [review]
I like the new changes 👍  . It looks good to me. The merge conflicts need to get solved before merging.

**Review by hollandjg — APPROVED** [review]
Looks good to me!

**Pull request merged**